pkg/apis/proto/source/v1/udsource.proto

```diff
@@ -0,0 +1,142 @@
+syntax = "proto3";
+
+option go_package = "github.com/numaproj/numaflow-go/pkg/apis/proto/source/v1";
+
+import "google/protobuf/timestamp.proto";
+import "google/protobuf/empty.proto";
+
+package source.v1;
+
+service UserDefinedSource {
+  // Read returns a stream of datum responses.
+  // The size of the returned ReadResponse is less than or equal to the num_records specified in ReadRequest.
+  // If the request timeout is reached on server side, the returned ReadResponse will contain all the datum responses that have been read.
```

Comment:
```suggestion
  // If the request timeout is reached on server side, the returned ReadResponse will contain all the datum responses that have been read (which could be an empty list).
```

Reply:
Discussed this offline, when it is an empty list, should we consider adding another configuration in numaflow to see if continuously reading empty data should raise an alert?